src/main/java/org/spongepowered/common/mixin/core/world/MixinExplosion.java

```diff
@@ -56,11 +56,21 @@
     @SuppressWarnings("rawtypes")
     @Shadow public List affectedBlockPositions;
 
+    private boolean shouldBreakBlocks;
```

Comment:
this should be settable from the `ExplosionBuilder` as well, not just default to whether if `isSmoking == true`.